src/client/Microsoft.Identity.Client/Platforms/netdesktop/NetDesktopPlatformProxy.cs

```diff
@@ -120,7 +120,10 @@ protected override string InternalGetDeviceModel()
         /// <inheritdoc/>
         protected override string InternalGetOperatingSystem()
         {
-            return DesktopOsHelper.GetWindowsVersionString();
+            return IsWindows ?
+                AbstractPlatformProxy.WindowsOS                
+                : Environment.OSVersion.Platform.ToString(); // probably for Mono, which we don't support
```

Comment:
Why  does Mono matter here? Shouldn't NetDesktopPlatformProxy be only used on net462 now?

Reply:
It's just wierd that we have `IsWindows` helper on net462 tfm. But it exists and it should be used due to Mono. When I'm surprised, I add a comment.